python/cugraph/cugraph/tests/mg/test_mg_uniform_neighbor_sample.py

```diff
@@ -250,3 +256,34 @@ def test_mg_uniform_neighbor_sample_unweighted(dask_client):
     actual_dst = sampling_results.destinations
     actual_dst = actual_dst.compute().to_arrow().to_pylist()
     assert sorted(actual_dst) == sorted(expected_dst)
+
+
+def test_mg_uniform_neighbor_sample_ensure_no_duplicates(dask_client):
+    # See issue #2760
+    # This ensures that the starts are properly distributed
+
```

Comment:
I think this case will always pass on CI as I believe we run CI on a single GPU  ?  Just confirming you ran this on multiple GPUs and it indeed fix the error. 

Looks good other-wise.

Reply:
I can confirm I did run the MG tests, including the one based on your MRE.

Reply:
All passed

Reply:
Thanks for confirming @alexbarghi-nv